pkg/reconciler/certificate/certificate.go

```diff
@@ -111,6 +116,7 @@ func (c *Reconciler) reconcile(ctx context.Context, knCert *v1alpha1.Certificate
 
 	logger.Infof("Reconciling Cert-Manager certificate for Knative cert %s/%s.", knCert.Namespace, knCert.Name)
 	cmConfig := config.FromContext(ctx).CertManager
+	c.setHTTP01Challenges(knCert)
```

Comment:
Are we planning to delete the challenge information once it is fulfilled?

Reply:
After the challenge has succeed and the certificate has been issued cert-manager will delete the solver service. When this runs again `setHTTP01Challenges` won't find a matching solver service so `Status.HTTP01Challenges` will be empty.

Reply:
Should we just simply delete `HTTP01Challenges` from `Status` when the certificate is ready?
If cert-manager fails to delete the HTTP01 challenge resources for some reason, we don't need to still keep HTT01 challenge info in the Status when the certificate is already ready. WDYT?

Reply:
I guess that's the safest thing to do. If the challenge info doesn't get removed from the cert then ingress rules won't either and that's probably not something we want.